src/mapper/pkg/resolvers/helpers.go

```diff
@@ -64,10 +64,11 @@ func IntentsHolderConfigFromViper() (IntentsHolderConfig, error) {
 }
 
 type IntentsHolder struct {
-	store  intentsHolderStore
-	lock   sync.Mutex
-	client client.Client
-	config IntentsHolderConfig
+	store          intentsHolderStore
+	lock           sync.Mutex
+	client         client.Client
+	config         IntentsHolderConfig
+	newIntentFound bool
```

Comment:
I would do that with a variable that stores the last change time, and make the thing that tracks the intents changes monitor this value. It's just when it's a boolean you have this doubt about how the IntentsHolder implements this boolean, when a timestamp is a bit more explicit and generic.